### ISSUE TITLE
[Feature] Single Editor per Script when opening it in Dashboard

### ISSUE THREAD

**Issue by cyfung1031** [violentmonkey/violentmonkey#2000]
**Motivation**

Currently if we click the edit code `</>` button in extension popup, it will switch to the opened editor tab. otherwise create a new editor tab.

However, afterwards, if we open the dashboard and click the edit `</>` button in dashboard, it will open in that dashboard.

1. The dashboard disappears.
2. Another editor is opened.

So it become two editors for the same script, and no dashboard

**Proposed Solution**

As there is another an editor tab for that script, dashboard tab should keep unchanged, and just switch to that editor tab.

However, I am not sure that, there is no editor, should the editor to be opened in a new tab or in the current tab.

I think it can be a prompt to the user to confirm "do you want to open it in a new tab" etc. Or a persistent setting stored in VM for "always new tab" option.

**Use Cases**

**Comment by tophf** [comment]
I thought about doing it, but couldn't find a good solution for the UI. I guess we can open the editors as tabs under `Installed scripts` in wide layout or next to it in the compact layout.

Note that since we use standard `<a>` links you can already do it by holding modifier keys when click the edit link:
* <kbd>Ctrl</kbd><kbd>Shift</kbd>+click = a new active tab
* <kbd>Ctrl</kbd>+click = a new inactive tab
* <kbd>Shift</kbd>+click = a new active window

**Comment by cyfung1031** [comment]
>  I guess we can open the editors as tabs under `Installed scripts` in wide layout or next to it in the compact layout.

Guess we can share the tab bar with editor.

![Screen Shot 2024-01-10 at 2 38 39 (1)](https://github.com/violentmonkey/violentmonkey/assets/44498510/bd6eca54-82b2-4f3d-8b13-f6dc6a80c5d9)

Dashboard button is always here.   Users can click it to go back dashboard. "Close" will close the tab.

<img width="1468" alt="Screen Shot 2024-01-10 at 2 49 18" src="https://github.com/violentmonkey/violentmonkey/assets/44498510/6111c663-72f4-4c1e-85bc-e51e40f2307d">


popup menu click will always open a new browser tab if no editor editing the same file. switch to it if any.
dashboard edit will always open in the same browser tab, even there is another editor editing the same file.

**Comment by tophf** [comment]
Nah, we'll add editor tabs to our panel on the left. And maybe add `<<` button to toggle it between normal width and collapsed to just an icon.

> "Close" will close the browser tab

No, this is weird because the tab has a built-in closing mechanism.
